src/db.rs

```diff
@@ -3534,10 +3352,80 @@ mod tests {
             compactor_options,
             compression_codec: None,
             object_store_cache_options: ObjectStoreCacheOptions::default(),
-            block_cache: None,
             garbage_collector_options: None,
-            clock,
             default_ttl: ttl,
+            ..Default::default()
         }
     }
+
+    #[tokio::test]
+    async fn test_builder_api() {
```

Comment:
while I appreciate adding unit test coverage for PRs, I think this isn't necessary -- the test cases are covered by tests that test specifically the functionality related to the configuration.

I think we should remove this test since it's a little confusing (what does it actually test? that no error is thrown for most of them?)

Reply:
I agree. I wrote it mostly to play with the API at the beginning. Deleted